test/integration/ingress_webhook_test.go

```diff
@@ -0,0 +1,362 @@
+		invalidRegexInIngressPathTestCase(`invalid regex: '/foo[[['`),
+		testCaseIngressValidation{
+			Name: "path should start with '/' or '~/' (regex path) (Kong Gateway requirement for non-expressions router)",
+			Ingress: &netv1.Ingress{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: uuid.NewString(),
+					Annotations: map[string]string{
+						annotations.IngressClassKey: consts.IngressClass,
+					},
+				},
+				Spec: netv1.IngressSpec{
+					Rules: []netv1.IngressRule{
+						{
+							IngressRuleValue: netv1.IngressRuleValue{
+								HTTP: &netv1.HTTPIngressRuleValue{
+									Paths: []netv1.HTTPIngressPath{
+										constructIngressPathImplSpecific("/bar"),
+										constructIngressPathImplSpecific("/~foo[1-9]"),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			WantCreateErr:          true,
+			WantCreateErrSubstring: `should start with: / (fixed path) or ~/ (regex path)`,
+		},
+	)
+	testIngressValidationWebhook(ctx, t, namespace, testCases)
+}
+
+func TestIngressValidationWebhookExpressionsRouter(t *testing.T) {
+	skipTestForNonKindCluster(t)
+	skipTestForRouterFlavors(t, traditional, traditionalCompatible)
+
+	ctx := context.Background()
+	namespace := setUpEnvForTestingIngressValidationWebhook(ctx, t)
+	testCases := append(
+		commonIngressValidationTestCases(),
+		invalidRegexInIngressPathTestCase("regex parse error:\n    ^/foo[[[\n           ^\nerror: unclosed character class"),
+		testCaseIngressValidation{
+			Name: "valid regex path passes validation",
+			Ingress: &netv1.Ingress{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: uuid.NewString(),
+				},
+				Spec: netv1.IngressSpec{
+					IngressClassName: lo.ToPtr(consts.IngressClass),
+					Rules: []netv1.IngressRule{
+						{
+							IngressRuleValue: netv1.IngressRuleValue{
+								HTTP: &netv1.HTTPIngressRuleValue{
+									Paths: []netv1.HTTPIngressPath{
+										constructIngressPathImplSpecific("/bar"),
+										constructIngressPathImplSpecific("/~baz[1-9]"),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			WantCreateErr: false,
+		},
+		testCaseIngressValidation{
+			Name: "invalid regex path fails validation",
+			Ingress: &netv1.Ingress{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: uuid.NewString(),
+				},
+				Spec: netv1.IngressSpec{
+					IngressClassName: lo.ToPtr(consts.IngressClass),
+					Rules: []netv1.IngressRule{
+						{
+							IngressRuleValue: netv1.IngressRuleValue{
+								HTTP: &netv1.HTTPIngressRuleValue{
+									Paths: []netv1.HTTPIngressPath{
+										constructIngressPathImplSpecific("/bar"),
+										constructIngressPathImplSpecific("/~baz[1-9]"),
+									},
+								},
+							},
+						},
+						{
+							IngressRuleValue: netv1.IngressRuleValue{
+								HTTP: &netv1.HTTPIngressRuleValue{
+									Paths: []netv1.HTTPIngressPath{
+										constructIngressPathImplSpecific("/~foo[[["),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
```

Comment:
I think that we could make the test cases more concise if we used a builder for constructing Ingress. This one could be e.g.:

```go
Ingress: NewIngressBuilder().
  WithRule(
    constructIngressPathImplSpecific("/bar"),
    constructIngressPathImplSpecific("/~baz[1-9]"),
  ).
  WithRule(
    constructIngressPathImplSpecific("/~foo[[[")),
  ).
  Build()
```

I omitted the default ingress class and name as they're almost always the same. In the case of a test with `third-party-ingress-class` we could use `WithIngressClassName`.

Reply:
I can create such a builder because for Ingress we don't have such